mlx/ops.cpp

```diff
@@ -453,8 +453,10 @@ array flatten(const array& a, StreamOrDevice s /* = {} */) {
 
 array hadamard_transform(
     const array& a,
-    float scale /* = 1.0 */,
+    std::optional<float> scale_ /* = std::nullopt */,
     StreamOrDevice s /* = {} */) {
+  // Default to an orthonormal Hadamard matrix scaled by 1/sqrt(N)
+  float scale = scale_.value_or(1.0f / std::sqrt(a.shape(-1)));
```

Comment:
Nitty nit: ternaries are nicere here because only one of the sides is evaluated:

```
float scale = scale_.has_value ? *scale_ : std::sqrt(a.shape(-1)))
```